src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs

```diff
@@ -329,21 +329,18 @@ public static void ValidateUtf8(ReadOnlySpan<byte> utf8Buffer)
 #else
             try
             {
-#if NET
-                s_utf8Encoding.GetCharCount(utf8Buffer);
```

Comment:
Why do you think this is dead code? We generally try to use safe alternative APIs whenever they are available.

Reply:
You have a `#if NET` inside of an `#else` of an `#if NET`.

These used to be separate targets for older TFMs (e.g. for net7 and net8), but now you just have multiple nested checks with conflicting conditions.